provers/risc0/driver/src/bonsai.rs

```diff
@@ -202,6 +218,10 @@ pub async fn prove_bonsai<O: Eq + Debug + DeserializeOwned>(
         assumption_uuids.clone(),
     )?;
 
+    if let Some(write) = write {
+        write.store_id(proof_key, session.uuid.clone())?;
```

Comment:
just thought the read/write is somehow misleading, especially this write, if not db, maybe <dyn idStore/taskStore> with writer/reader better.